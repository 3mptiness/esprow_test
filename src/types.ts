export type ChangeHandler = (change: {
	index: number;
	prop: string;
	value: string;
}) => void;

export type InputType =
	| 'input'
	| 'textarea'
	| 'email'
	| 'number'
	| 'date'
	| 'radio';

export type JsonPrimitive = string | number | boolean | null;
export type JsonValue = JsonPrimitive | JsonObject | JsonArray;
export type JsonObject = { [key: string]: JsonValue };
export type JsonArray = JsonValue[];

export type Primitive = string | number | boolean;
export type RowType = Record<string, string | number>;

export type TableProps = {
	jsonArray: JsonArray;
	onChange?: (data: JsonArray) => void;
	rowHeight?: number;
};

export type HeadersMap = Map<string, InputType>;

export type XY = { x: number; y: number };
