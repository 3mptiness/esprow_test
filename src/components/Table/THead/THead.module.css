.Column {
  cursor: pointer;
}

.Sort {
  position: absolute;
}
