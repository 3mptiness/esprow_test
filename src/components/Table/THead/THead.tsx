import React, { FC, memo } from 'react';
import { HeadersMap } from '../../../types';
import css from './THead.module.css';

type THeadProps = {
	headers: HeadersMap;
	getSortArrow: (title: string) => string;
	toggleSort: (title: string) => void;
};

export const THead: FC<THeadProps> = memo((props) => {
	const { headers, getSortArrow, toggleSort } = props;
	return (
		<thead>
			<tr>
				{Array.from(headers).map(([title]) => {
					const arrow = getSortArrow(title);
					return (
						<th
							key={title}
							onClick={() => toggleSort(title)}
							className={css.Column}
						>
							{title}
							{arrow && <span className={css.Sort}>{arrow}</span>}
						</th>
					);
				})}
			</tr>
		</thead>
	);
});
