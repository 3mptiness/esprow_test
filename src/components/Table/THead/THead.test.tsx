import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { THead } from './THead';
import { HeadersMap } from '../../../types';

describe('THead', () => {
	it('success THead render', () => {
		const headers: HeadersMap = new Map([
			['column', 'input'],
			['test', 'radio'],
		]);
		const getSortArrow = (title: string) => (title === 'test' ? 'arrow' : '');
		expect(
			createRenderer().render(
				<THead
					headers={headers}
					getSortArrow={getSortArrow}
					toggleSort={jest.fn()}
				/>
			)
		).toMatchSnapshot();
	});
});
