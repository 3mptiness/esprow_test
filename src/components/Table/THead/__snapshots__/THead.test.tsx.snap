// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THead success THead render 1`] = `
<thead>
  <tr>
    <th
      className="Column"
      onClick={[Function]}
    >
      column
      
    </th>
    <th
      className="Column"
      onClick={[Function]}
    >
      test
      <span
        className="Sort"
      >
        arrow
      </span>
    </th>
  </tr>
</thead>
`;
